### ISSUE TITLE
Robots.txt file can't be emptied/deleted from backend

### ISSUE THREAD

**Issue by pdanzinger** [magento/magento2#7098]
### Preconditions

1. I tested it with Magento 2.1.1 CE on Windows with php 5.6, but the issue is most likely platform independent
### Steps to reproduce

1. Go to Stores->Configuration->General->Design
2. Add content to "Edit custom instructions of robots.txt File" and save
3. Remove the content from the field and save again
### Expected result

1. robots.txt file gets deleted or emptied
### Actual result

1. Field stays filled with the old content, robots.txt remains

The reason seems to be the code inside the class for the robots config: Magento\Config\Model\Config\Backend\Admin\Robots:afterSave()

```
    public function afterSave()
    {
        if ($this->getValue()) {
            $this->_directory->writeFile($this->_file, $this->getValue());
        }

        return parent::afterSave();
    }
```

Replacing it with this code worked for me, but I'm not sure if it doesn't break on other systems:

```
    public function afterSave()
    {
        if ($this->getValue()) {
            $this->_directory->writeFile($this->_file, $this->getValue());
        } else {
            $this->_directory->delete($this->_file);
        }

        return parent::afterSave();
    }
```

**Comment by palamar** [comment]
Internal task MAGETWO-60065 has been created for this issue

**Comment by sambolek** [comment]
This can be traced:

- \Magento\Theme\Controller\Adminhtml\Design\Config\Save.php:72
- \Magento\Theme\Model\Design\Config\Storage.php:114
- \Magento\Theme\Model\Design\Config\ValueChecker.php:56
- \Magento\Theme\Model\Design\Config\ValueProcessor.php:46
- \Magento\Config\Model\Config\Backend\Admin\Robots:71

In essence, \Magento\Theme\Model\Design\Config\ValueProcessor->process() calls afterLoad method on backend source model of the input field. AfterLoad method of the source model sets default value from existing file, if value we're attempting to save **isn't a string** (if empty, the field is set to '', e.g. empty string).

AfterSave() in Robots source model never even gets called, as it detects no changes were done to the field.

--

The way I see it, there are a few solutions to the issue:

1. Remove _afterLoad logic from the Robots model, which would mean that robots field wouldn't get values form robots.txt file either
2. Add "Remove robots file" button (possibly as AJAX action), so we can easily remove robots.txt file, regardless of input value
3. some other solution I didn't think of? 

I'm interested in solving this one, just not sure which way to proceed.

**Comment by magento-engcom-team** [comment]
@pdanzinger, thank you for your report.
The issue is already fixed in develop branch, 2.2.0
